app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt

```diff
@@ -0,0 +1,181 @@
+                    .padding(paddingValues)
+                    .verticalScroll(scrollState)
+                    .testTag("editProfileContent"),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              HorizontalDivider(color = Color.LightGray, thickness = 1.dp)
+              Spacer(modifier = Modifier.height(50.dp))
+              Box {
+                ProfilePicture(R.drawable.default_profile_picture) // TODO change to current image
+                Box(modifier = Modifier.align(Alignment.BottomEnd)) {
+                  CircleWithIcon(Icons.Filled.Edit, MaterialTheme.colorScheme.primary)
+                }
+              }
+              Spacer(modifier = Modifier.height(66.dp))
+              CustomInputField(
+                  value = name,
+                  onValueChange = { newName ->
+                    if (newName.length <= maxUsernameLength) {
+                      name = newName
+                    }
+                  },
+                  label = "Name",
+                  placeholder = "Current name",
+                  supportingText = "Max $maxUsernameLength characters",
+                  trailingIcon = Icons.Filled.Clear,
+                  modifier = Modifier.testTag("editProfileNameInput"))
+              Spacer(modifier = Modifier.height(16.dp))
+              Row {
+                CustomInputField(
+                    value = description,
+                    onValueChange = { newDescription ->
+                      if (newDescription.length <= maxDescriptionLength) {
+                        description = newDescription
+                      }
+                    },
+                    label = "Description",
+                    placeholder = "Current description",
+                    supportingText = "Max $maxDescriptionLength characters",
+                    singleLine = false,
+                    trailingIcon = Icons.Filled.Clear,
+                    modifier = Modifier.testTag("editProfileDescriptionInput"))
+              }
+              Spacer(modifier = Modifier.height(120.dp))
+              Box(
+                  modifier =
+                      Modifier.border(1.dp, PrimaryGradientBrush, RoundedCornerShape(30.dp))
+                          .width(320.dp)
+                          .height(48.dp)) {
+                    Button(
+                        onClick = { /* Handle button click */},
+                        modifier = Modifier.fillMaxWidth().testTag("saveProfileButton"),
+                        colors =
+                            ButtonDefaults.buttonColors(containerColor = lightThemeBackground)) {
+                          Text(
+                              text = "Save",
+                              fontWeight = FontWeight.Bold,
+                              fontSize = 16.sp,
+                              color = MaterialTheme.colorScheme.primary)
+                        }
+                  }
+            }
```

Comment:
We have a reusable **PrincipalButton** composable in the **MainComponents** file, designed for this type of commonly used button in the app. Consider using it here to reduce code duplication.